src/Transport/Seam/Dispatcher.cs

```diff
@@ -46,11 +49,30 @@ Task DispatchBatches(IList<Batch> outgoingBatches, BrokeredMessageReceiveContext
             // default behavior is to postpone sends until complete (and potentially complete them as a single tx if possible)
             // but some messages may need to go out immediately
 
-            var toBeDispatchedImmediately = outgoingBatches.Where(t => t.RequiredDispatchConsistency == DispatchConsistency.Isolated);
-            var toBeDispatchedOnComplete = outgoingBatches.Where(t => t.RequiredDispatchConsistency == DispatchConsistency.Default);
+            return DispatchAccordingToIsolationLevel(outgoingBatches, receiveContext);
+        }
+
+        async Task DispatchAccordingToIsolationLevel(IList<Batch> outgoingBatches, BrokeredMessageReceiveContext receiveContext)
+        {
+            var batchesWithIsolatedDispatchConsistency = outgoingBatches.Where(t => t.RequiredDispatchConsistency == DispatchConsistency.Isolated);
+            var batchesWithDefaultConsistency = outgoingBatches.Where(t => t.RequiredDispatchConsistency == DispatchConsistency.Default);
+
+            await routeOutgoingBatches.RouteBatches(batchesWithIsolatedDispatchConsistency, receiveContext).ConfigureAwait(false);
+            await DispatchWithTransactionScopeIfRequired(batchesWithDefaultConsistency, receiveContext).ConfigureAwait(false);
```

Comment:
If we can get multiple dispatcher invocations for isolated consistency messages, and only one invocation for default (batched) level, why do we call `DispatchWithTransactionScopeIfRequired()` each time? Should we first see if it's needed? Or within the method only invoke a router if collection contains any messages?